src/element/mod.rs

```diff
@@ -459,6 +504,13 @@ impl Element {
         self.as_int().ok_or_else(|| self.expected(IonType::Int))
     }
 
+    pub fn try_into_int(self) -> ControlFlow<Element, Int> {
```

Comment:
I think you're using `ControlFlow` everywhere to avoid needing to define another error type to hand back the `Element` in the error case. Is that right? If so, I think there's enough methods like this that we could justify a `CastResult` or `ConversionResult` to make things more self-explanatory.

Reply:
Essentially, yes. I'm not wedded to it's use and open to replace it.

Using something like `Result<Sequence, Element>` for the `try_into_sequence` didn't feel quite right, as using `?` wasn't quite as expected.

`ControlFlow` has `FromResidual` experimental implementation. But I suppose that could always be added for a custom type.